### ISSUE TITLE
NoSuchMethodError: The method 'refresh' was called on null.

### ISSUE THREAD

**Issue by DisDis** [dart-lang/sdk#36419]
Analyzer Feedback from IntelliJ

## Version information

- `IDEA WS-191.6183.63`
- `2.2.0`
- `WS-191.6183.63, JRE 1.8.0_202-release-1483-b39x64 JetBrains s.r.o, OS Linux(amd64) v4.15.0-46-generic, screens 3840x1080`

## Exception
```
Dart analysis server, SDK version 2.2.0, server version 1.23.0, error: Captured exception
NoSuchMethodError: The method 'refresh' was called on null.
Receiver: null
Tried calling: refresh(Instance of 'DeclarationsContext')
#0      Object.noSuchMethod (dart:core/runtime/libobject_patch.dart:50:5)
#1      DeclarationsTracker._performChangeFile (package:analyzer/src/services/available_declarations.dart:621:15)
#2      DeclarationsTracker.doWork (package:analyzer/src/services/available_declarations.dart:483:7)
#3      CompletionLibrariesWorker.performWork (package:analysis_server/src/domains/completion/available_suggestions.dart:171:13)
#4      _AsyncAwaitCompleter.start (dart:async/runtime/libasync_patch.dart:49:6)
#5      CompletionLibrariesWorker.performWork (package:analysis_server/src/domains/completion/available_suggestions.dart:170:27)
#6      AnalysisDriverScheduler._run (package:analyzer/src/dart/analysis/driver.dart:2125:35)
#7      _asyncThenWrapperHelper.<anonymous closure> (dart:async/runtime/libasync_patch.dart:77:64)
#8      _rootRunUnary (dart:async/zone.dart:1132:38)
#9      _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#10     _FutureListener.handleValue (dart:async/future_impl.dart:126:18)
#11     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:639:45)
#12     Future._propagateToListeners (dart:async/future_impl.dart:668:32)
#13     Future._complete (dart:async/future_impl.dart:473:7)
#14     _SyncCompleter.complete (dart:async/future_impl.dart:51:12)
#15     _AsyncAwaitCompleter.complete (dart:async/runtime/libasync_patch.dart:28:18)
#16     _completeOnAsyncReturn (dart:async/runtime/libasync_patch.dart:294:13)
#17     Monitor.signal (package:analyzer/src/dart/analysis/status.dart)
#18     _asyncThenWrapperHelper.<anonymous closure> (dart:async/runtime/libasync_patch.dart:77:64)
#19     _rootRunUnary (dart:async/zone.dart:1132:38)
#20     _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#21     _FutureListener.handleValue (dart:async/future_impl.dart:126:18)
#22     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:639:45)
#23     Future._propagateToListeners (dart:async/future_impl.dart:668:32)
#24     Future._addListener.<anonymous closure> (dart:async/future_impl.dart:348:9)
#25     _rootRun (dart:async/zone.dart:1124:13)
#26     _CustomZone.run (dart:async/zone.dart:1021:19)
#27     _CustomZone.runGuarded (dart:async/zone.dart:923:7)
#28     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:963:23)
#29     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
#30     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
#31     _Timer._runTimers (dart:isolate/runtime/libtimer_impl.dart:391:30)
#32     _Timer._handleMessage (dart:isolate/runtime/libtimer_impl.dart:416:5)
#33     _RawReceivePortImpl._handleMessage (dart:isolate/runtime/libisolate_patch.dart:171:12)
```

For additional log information, please append the contents of
file:///tmp/report1.txt.

**Comment by johnpryan** [comment]
this seems to happen when running a flutter app for the first time

**Comment by kevmoo** [comment]
@devoncarew @stereotype441 – seems to be reported A LOT!

**Comment by stereotype441** [comment]
@scheglov can you take a look?

**Comment by scheglov** [comment]
I think this was fixed in cc1ddfb3372245edf04f6c84978528fefd6d0e38 as https://github.com/dart-lang/sdk/issues/36212.

**Comment by kevmoo** [comment]
Duplicate of https://github.com/dart-lang/sdk/issues/36212 – which was fixed in https://github.com/dart-lang/sdk/commit/cc1ddfb3372245edf04f6c84978528fefd6d0e38
